release-23.1: c2c: allow user to pass start time as a decimal to crdb_internal.fingerprint()

msbutler opened the pull request:
Backport 1/1 commits from #104219.

/cc @cockroachdb/release

---

Previously, the user could only pass the startTime variable to crdb_internal.fingerprint() as a TimestampTZ, which only has microsecond precision. To enable nanosecond precision, this patch allows the user to pass the startTime as a decimal. This patch is consistent with the user's ability to pass and AOST timestamp as a decimal as well.

Informs #103072

Release note: None

Release justification: test infra change

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/104794)
<!-- Reviewable:end -->

msbutler committed:
c2c: allow user to pass start time as a decimal to crdb_internal.fingerprint()

Previously, the user could only pass the startTime variable to
crdb_internal.fingerprint() as a TimestampTZ, which only has microsecond
precision. To enable nanosecond precision, this patch allows the user to pass
the startTime as a decimal. This patch is consistent with the user's ability to
pass and AOST timestamp as a decimal as well.

Informs #103072

Release note: None

msbutler commented:
note that for the backport, i had to change how we parse the decimal timestamp, because ` tree.AsDDecimal(args[1])` isn't on 23.1

lidorcarmel commented:
> note that for the backport, i had to change how we parse the decimal timestamp, because ` tree.AsDDecimal(args[1])` isn't on 23.1

did you consider backporting `AsDDecimal`?

msbutler commented:
I did not, as DR doesn't own `AsDDecimal`.  I didn't want to increase the surface area of this backport.
